components/ui/PortalPopup.tsx

```diff
@@ -0,0 +1,228 @@
+
+    // eslint-disable-next-line react/no-find-dom-node
+    const toggleNode = findDOMNode(this);
+
+    if (toggleNode && toggleNode instanceof Element && this.popupNode.current) {
+      const toggleRect = toggleNode.getBoundingClientRect();
+      const { width, height } = this.popupNode.current.getBoundingClientRect();
+      let left = 0;
+      let top = 0;
+
+      switch (this.getPlacement()) {
+        case PopupPlacement.Bottom:
+          left = toggleRect.left + toggleRect.width / 2 - width / 2;
+          top = toggleRect.top + toggleRect.height;
+          break;
+        case PopupPlacement.BottomLeft:
+          left = toggleRect.left;
+          top = toggleRect.top + toggleRect.height;
+          break;
+        case PopupPlacement.BottomRight:
+          left = toggleRect.left + toggleRect.width - width;
+          top = toggleRect.top + toggleRect.height;
+          break;
+        case PopupPlacement.LeftTop:
+          left = toggleRect.left - width;
+          top = toggleRect.top;
+          break;
+        case PopupPlacement.RightTop:
+          left = toggleRect.left + toggleRect.width;
+          top = toggleRect.top;
+          break;
+        case PopupPlacement.RightBottom:
+          left = toggleRect.left + toggleRect.width;
+          top = toggleRect.top + toggleRect.height - height;
+          break;
+        case PopupPlacement.TopLeft:
+          left = toggleRect.left;
+          top = toggleRect.top - height;
+          break;
+      }
+
+      // save width and height (and later set in `render`) to avoid resizing the popup element,
+      // when it's placed close to the window edge
+      this.setState({
+        left: window.pageXOffset + left,
+        top: window.pageYOffset + top,
+        width,
+        height,
+      });
+    }
+  };
+
+  renderActual = ({ leftFix = 0, topFix = 0 }) => {
+    const { className, overlay, noPadding } = this.props;
+    const placement = this.getPlacement();
+    const style = isMeasured(this.state)
+      ? {
+          left: this.state.left + leftFix,
+          top: this.state.top + topFix,
+          width: this.state.width,
+          height: this.state.height,
+        }
+      : undefined;
+
+    return (
+      <Popup
+        arrowStyle={
+          isMeasured(this.state)
+            ? this.adjustArrowPosition(placement, { leftFix, topFix })
+            : undefined
+        }
+        className={className}
+        noPadding={noPadding}
+        placement={placement}
+        ref={this.popupNode}
+        style={style}>
+        {overlay}
+      </Popup>
+    );
+  };
+
+  render() {
+    return (
+      <>
+        {this.props.children}
+        {this.props.overlay && (
+          <PopupPortal>
+            <ScreenPositionFixer ready={isMeasured(this.state)}>
+              {this.renderActual}
+            </ScreenPositionFixer>
+          </PopupPortal>
+        )}
+      </>
+    );
+  }
+}
+
+class PopupPortal extends React.Component {
```

Comment:
Maybe we can find a better name for this? I found confusing having PortalPopup and PopupPortal classes here. 

Otherwise, It looks good to me!

Reply:
Ah yeah indeed, I didn't notice. Maybe I could rename it to something like: "PortalWrapper" or "PortalPopupWrapper", would it be better ? Or do you have something in mind?

Reply:
Yes, I would say `PortalWrapper` fits perfectly here